apps/api/src/services/user-manager.service.ts

```diff
@@ -23,10 +25,13 @@ export class UserManager {
   private userWalletCache: Map<string, string>; // walletAddress -> userId
   // Cache for user profiles by ID
   private userProfileCache: Map<string, SelectUser>; // userId -> user profile
+  // Email service for sending verification emails
+  private emailService: EmailService;
 
   constructor() {
     this.userWalletCache = new Map();
     this.userProfileCache = new Map();
+    this.emailService = new EmailService();
```

Comment:
instead of instantiating the service in the constructor, we should have the constructor take it as a param and then pass it to `UserManager` the root `services/index.ts` instance.